features/steps/static_primary.py

```diff
@@ -0,0 +1,25 @@
+import json
+import patroni.psycopg as pg
+
+from behave import step, then
+from time import sleep, time
+
+
+@step('I start {name:w} as static primary')
+def start_patroni_with_static_primary(context, name):
+    return context.pctl.start(name, custom_config={'bootstrap': {'dcs': {'static_primary': name}}})
+
+
+@step('I start {name:w} with a configured static primary will not boot after {time_limit:d} seconds')
+def start_patroni_as_replica_with_static_primary(context, name, time_limit):
+    return context.pctl.start_with_expected_failure(name, max_wait_limit=time_limit)
+
+
+@step('"{name}" key not in DCS after waiting {time_limit:d} seconds')
+def check_member_not_present(context, name, time_limit):
+    sleep(time_limit)
+    try:
+        json.loads(context.dcs_ctl.query(name))
+        assert False, "found value under DCS key {} after {} seconds".format(name, time_limit)
```

Comment:
Meh, where is the actual check?

Reply:
The check is the call to `json.loads`. If it succeeds, then this indicates that we found a member (`name`) and it was not supposed to be there.

Reply:
The thing is that the running Patroni continues to call touch_member() even if Postgres isn't running.
Therefore, the member key never expires.

The `assert` raises the exception which is just eaten on the next line. Or in other words, we wasted `time_limit` just for nothing.